src/ucp/api/ucp.h

```diff
@@ -3304,32 +3329,40 @@ ucp_atomic_fetch_nb(ucp_ep_h ep, ucp_atomic_fetch_op_t opcode,
 
 /**
  * @ingroup UCP_COMM
- * @brief Post an atomic fetch operation.
+ * @brief Post an atomic memory operation.
  *
- * This routine will post an atomic fetch operation to remote memory.
+ * This routine will post an atomic operation to remote memory.
  * The remote value is described by the combination of the remote
  * memory address @a remote_addr and the @ref ucp_rkey_h "remote memory handle"
- * @a rkey.
- * The routine is non-blocking and therefore returns immediately. However the
- * actual atomic operation may be delayed. The atomic operation is not considered complete
- * until the values in remote and local memory are completed.
- *
- * @note The user should not modify any part of the @a buffer or @a result after
- *       this operation is called, until the operation completes.
- * @note Only ucp_dt_make_config(4) and ucp_dt_make_contig(8) are supported in
- *       @a param->datatype, see @ref ucp_dt_make_contig
+ * @a rkey. The routine is non-blocking and therefore returns immediately.
+ * However the actual atomic operation may be delayed.
+ * If the reply buffer is not provided in @a param->reply_buffer, request handle
+ * is not returned and a user must call @ref ucp_ep_flush_nb or
+ * @ref ucp_worker_flush_nb to guarantee that the remote value has been updated.
+ * Otherwise, if fetch type of operation is being requested, the atomic
```

Comment:
Let's explicitly list (table 1) all supported fetch operation. table2: non-fetch operations. This is in addition to enum reference. What happens if fetch operation was requested but no reply buffer was provided ?

Reply:
@tonycurtis there is major change in the function behavior and description. please review.

Reply:
3339: the request handle
3340: "the user" sounds better
3342: if a "fetch" operation...
3343: in both

Reply:
@shamisp, we have the same op codes for fetch and post ops. If we provide reply_buffer, it will be a fetch, otherwise post